File: frontend/src/types/houseMember.ts
import type { User } from './user';

export type HouseMemberRelation = 'owner' | 'co-resident';
export type HouseMemberStatus = 'active' | 'removed';

export interface HouseMember {
  id: number;
  building: string;
  unit: string;
  room: string;
  userId: number;
  relation: HouseMemberRelation;
  invitedBy?: number;
  status: HouseMemberStatus;
  createdAt: string;
  user?: User;
}

File: backend/src/main/java/com/smartestate/entity/HouseMember.java
package com.smartestate.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("house_members")
public class HouseMember {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String building;
    private String unit;
    private String room;
    private Long userId;
    private String relation;
    private Long invitedBy;
    private String status;
    private LocalDateTime createdAt;

    @TableField(exist = false)
    private User user;
}
